modules/registration/import.js

```diff
@@ -0,0 +1,92 @@
+/* jshint browser: true */
+/* jshint -W097 */
+/* globals apiRequest, menubarElement */
+/* exported */
+
+function load() {
+  apiRequest('GET', 'registration/admin', null)
+    .then(function() {
+      new menubarElement(
+        {
+          'text': 'Registration Admin',
+          'function': 'registration/admin',
+          'title': 'Registration Administration',
+          'selectedStyle': 'event-color-primary',
+          'baseStyle': 'UI-yellow',
+          'icon': 'far fa-address-card'
+        }
+      ).createAdminElement();
+    })
+    .catch(function(response) {
+      if (response instanceof Error) { throw response; }
+    });
+
+  apiRequest('GET', 'registration/ticket/list', 'include=event')
+    .then(function(response) {
+      var result = JSON.parse(response.responseText);
+      var data = null;
+      if (Array.isArray(result.data)) {
+        data = result.data;
+      } else {
+        data = Array(result);
+      }
+      if (data.length > 0) {
+        var eventName =  data[0].event.name;
+        var generated = 0
+        data.forEach(function(ticket) {
+          if (ticket.BoardingPassGenerated) {
+            generated += 1;
+          }
+        });
+        if (generated < data.length) {
+          apiRequest('GET', 'registration/open', null)
```

Comment:
Once we get into really long promise chains, I like to actually pull the functions up and name them, then it can be

```javascript
foo()
  .then(bar())
  .then(baz())
  .then(quux())
  .catch(everythingIsFucked());
```

Reply:
Most of the long promise chains I am runngin into are more
foo()
 .then( function() {
    .then( function() {
     })
 })

So nested instead of sequential.  Unless I am misunderstanding it we end up needing to next them like that.